drivers/storage/portworx/portworx.go

```diff
@@ -467,6 +467,56 @@ func (p *portworx) getCurrentMaxStorageNodesPerZone(
 	return 0, fmt.Errorf("storage disabled")
 }
 
+func (p *portworx) preflightShouldRun(toUpdate *corev1.StorageCluster) bool {
+
+	if !pxutil.IsFreshInstall(toUpdate) { // Preflight should only run freshInstall
+		return false
+	}
+
+	clusterPXver := pxutil.GetPortworxVersion(toUpdate)
+
+	pxVer30, _ := version.NewVersion("3.0")
+	if !clusterPXver.GreaterThanOrEqual(pxVer30) { // Preflight should only run if the PX version is 3.0.0 and above
+		return false
+	}
+
+	check, ok := toUpdate.Annotations[pxutil.AnnotationPreflightCheck]
+	check = strings.TrimSpace(strings.ToLower(check))
+	if ok && check == "true" { // Preflight is enabled
+		return true
+	}
+
+	if !preflight.RequiresCheck() { // Preflight is only supported on AWS, VSPHERE & Pure
+		return false
+	}
+
+	if preflight.IsAWS() { // Preflight runs on AWS & PX 3.0.0 or above
+		return true
+	}
+
+	pxVer31, _ := version.NewVersion("3.1")
+	if clusterPXver.GreaterThanOrEqual(pxVer31) { // PX version is 3.1.0 or above
+		if pxutil.IsPure(toUpdate) { // Preflight runs on Pure & PX 3.1.0 or above
+			return true
+		}
+
+		if pxutil.IsVsphere(toUpdate) {
+			if preflight.IsPKS() { // Don't run preflight on Vsphere w/PKS
+				return false
```

Comment:
Let's log these false cases.

Reply:
I will add logging and see but what it looks like but this routine gets executed in the reconcile loop so its called every 20s so not sure we should log these.    It may end up spamming the logs.  But will give it a try.